platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -523,17 +526,23 @@ private void logout() {
   }
 
   private void logSecurityAuditRole() {
-    if (shouldAuditSubject()) {
-      SecurityLogger.audit(
-          "Subject with admin privileges has logged out: {}",
-          getIdpSecurityAssertion().getPrincipal().getName());
+    SecurityToken idpSecToken = getIdpSecurityToken();
+    if (idpSecToken != null) {
```

Comment:
Twice we check to see if idpSpecToken is null. I'd like to see that encapsulated in a method with a name meaningful name. I don't know enough about the issue to suggest a good name -- maybe "isSessionMissingToken" or "requestHasToken"?

Reply:
I personally like how it is now because it seems a bit more straightforward, it is only twice, and they are handled differently in the sense that one of them simply checks to see if we should continue and the other returns a `buildLogoutResponse`

Reply:
What does it mean when the token is null?

Reply:
If the token is null it could mean that the saml assertion was removed from the session. The previous code was calling this in a way that lead to a null pointer exception.

Reply:
I think both ideas are great!

However, what would persuade me that encapsulating would be the way to go here, is if the condition we are checking was more complex. Since we are only checking that `idpSecToken` is null, that is simple enough to have inline.